go/tools/gazelle/gazelle/main.go

```diff
@@ -33,15 +33,12 @@ import (
 )
 
 var (
-	goPrefix  = flag.String("go_prefix", "", "go_prefix of the target workspace")
-	repoRoot  = flag.String("repo_root", "", "path to a directory which corresponds to go_prefix, otherwise gazelle searches for it.")
-	mode      = flag.String("mode", "fix", "print: prints all of the updated BUILD files\n\tfix: rewrites all of the BUILD files in place\n\tdiff: computes the rewrite but then just does a diff")
-	buildName = flag.String("build_name", "BUILD", "name of output build files to generate, defaults to 'BUILD'")
+	buildFileName = flag.String("build_file_name", "BUILD", "name of output build files to generate, defaults to 'BUILD'")
+	goPrefix      = flag.String("go_prefix", "", "go_prefix of the target workspace")
+	repoRoot      = flag.String("repo_root", "", "path to a directory which corresponds to go_prefix, otherwise gazelle searches for it.")
+	mode          = flag.String("mode", "fix", "print: prints all of the updated BUILD files\n\tfix: rewrites all of the BUILD files in place\n\tdiff: computes the rewrite but then just does a diff")
 
-	validBuildNames = map[string]bool{
-		"BUILD":       true,
-		"BUILD.bazel": true,
-	}
+	buildFileNames = []string{"BUILD.bazel", "BUILD"}
```

Comment:
Exactly. The iteration order over Go maps is undefined (and is indeed randomized) so the old code would have undefined behaviour if both files were present. I believe that Bazel itself prioritises `BUILD.bazel` over `BUILD`.